internal/onprem/hydrate.go

```diff
@@ -135,30 +128,30 @@ func (h *OnPremEventsHydrator) extractEvents(filename string, msg kafka.Message)
 func (h *OnPremEventsHydrator) notifyEvent(envelope types.EventEnvelope) {
 	h.logger.WithFields(logrus.Fields{
 		"cluster_id": envelope.Key,
-	}).Debug("notfiying event for on-prem cluster")
+	}).Debug("notifying event for on-prem cluster")
 	if err := h.writer.Write(h.ctx, envelope.Key, envelope.Event); err != nil {
-		h.logger.WithFields(logrus.Fields{
-			"err": err,
-		}).Warning("error when notifying event")
+		h.logger.WithError(err).Warning("error when notifying event")
 	}
 }
 
 func (h *OnPremEventsHydrator) downloadURL(url string, msg kafka.Message) {
-	h.logger.WithFields(logrus.Fields{
-		"url": url,
-	}).Info("downloading file from url")
+	logger := h.logger.WithField("url", url)
+	logger.Info("downloading file from url")
+
 	downloadedFilename, err := h.downloader.DownloadFile(url)
 	if err != nil {
-		h.logger.WithFields(logrus.Fields{
-			"url": url,
-			"err": err,
-		}).Warning("error downloading from url")
+		logger.WithError(err).Warning("error downloading from url")
 		return
 	}
 	h.untarChannel <- FilenameMessage{Filename: downloadedFilename, Msg: msg}
 }
 
 func (h *OnPremEventsHydrator) ProcessMessage(ctx context.Context, msg *kafka.Message) error {
+	// This doesn't look right
```

Comment:
@rccrdpccl feedbacks apprecaited here :pray:

Reply:
Definitely wrong logic here, it should be:
* if no service key in all headers, commit offset: no need to re-parse this message as we don't even need to process it
* it could be committed if there is no header, as this message is not for us
* We will eventually commit the offset when the message will be processed. Maybe we should reorganize the code to make it clearer